Importing the all necessary libraries:

In [2]:
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
sns.set() # Apply the default seaborn theme, scaling, and color palette

import pandas as pd
import numpy as np

import warnings
warnings.filterwarnings("ignore")

This dataset appears to represent monthly electricity billing records of customers from a specific area. It contains 34 columns and includes detailed information about each customer's electricity usage, billing, and payments.

Load the datasets

In [1]:
import pandas as pd

# Load the CSV file into a DataFrame
df = pd.read_csv("D:\GitHub\M6.csv")



<>:4: SyntaxWarning: invalid escape sequence '\G'
<>:4: SyntaxWarning: invalid escape sequence '\G'
C:\Users\USER\AppData\Local\Temp\ipykernel_3788\3951625316.py:4: SyntaxWarning: invalid escape sequence '\G'
  df = pd.read_csv("D:\GitHub\M6.csv")
C:\Users\USER\AppData\Local\Temp\ipykernel_3788\3951625316.py:4: DtypeWarning: Columns (2,4,18,19,27) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("D:\GitHub\M6.csv")


## Display the 5 first  rows

In [2]:
df.head()

LOC              CUST_NAM F_H_NAME  CUST_NUM        A_C             ADDRESS  \
0  M6  MD.ALI  AKBAR  MOLLA      NaN  75900134  A/1632/77  NORTH  KOIRAKURI,,   
1  M6       MD.ABDUL  HAMID      NaN  75900149  A/1116/79  NORTH  KOIRAKURI,,   
2  M6           ABDUL  BARI      NaN  75900153  A/1178/80  NORTH  KOIRAKURI,,   
3  M6    MD.ISMAIL  HOSSAIN      NaN  75900168  A/1049/81  NORTH  KOIRAKURI,,   
4  M6       MD.MAFIZ  UDDIN      NaN  75900172  A/1368/82  NORTH  KOIRAKURI,,   

  BOOK  BG  WALK_SEQ METER_NO  ...  CUR_M_BILL             PAY_DATE  \
0  H02   2  1,320.00   611008  ...    1,574.94  05/11/2025 00:00:00   
1  H02   2  1,350.00   377600  ...      895.90  05/11/2025 00:00:00   
2  H02   2  1,360.00   557275  ...    1,379.70  05/12/2025 00:00:00   
3  H02   2  1,370.00   188176  ...      408.08  05/12/2025 00:00:00   
4  H02   2  1,380.00   203266  ...    1,463.68  05/12/2025 00:00:00   

   PRIN_PAID  VAT_PAID TOTAL_ADJ PRN_BALANCE  VAT_BALANCE LPS_BALANCE  \
0   1,270.00     64.00      0.06    1,499.94        75.00        0.00   
1     500.00     25.00      0.10      852.90        43.00        0.00   
2   1,289.00     65.00      0.30    1,313.70        66.00        0.00   
3     957.00     47.00     -0.08      389.08        19.00        0.00   
4   4,750.00    250.00      0.32    4,844.12       231.00      336.56   

  TOTAL_BALANCE TOTAL_BILL  
0      1,574.94   1,575.00  
1        895.90     896.00  
2      1,379.70   1,380.00  
3        408.08     408.00  
4      5,411.68   5,412.00  

[5 rows x 34 columns]

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16714 entries, 0 to 16713
Data columns (total 34 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   LOC            16714 non-null  object 
 1   CUST_NAM       16714 non-null  object 
 2   F_H_NAME       334 non-null    object 
 3   CUST_NUM       16714 non-null  int64  
 4   A_C            16713 non-null  object 
 5   ADDRESS        16714 non-null  object 
 6   BOOK           16714 non-null  object 
 7   BG             16714 non-null  int64  
 8   WALK_SEQ       16714 non-null  object 
 9   METER_NO       16714 non-null  object 
 10  ME_PHASE       14726 non-null  float64
 11  S_LOAD         16714 non-null  float64
 12  C_LOAD         16714 non-null  float64
 13  METER_COND     14726 non-null  float64
 14  OMF            16714 non-null  object 
 15  CATAGORY       16714 non-null  object 
 16  BILL_MONTH     16714 non-null  int64  
 17  C_SR_RED       13690 non-null  object 
 18  C_OFPK

 Variable Identification

In [3]:
df.dtypes

LOC               object
CUST_NAM          object
F_H_NAME          object
CUST_NUM           int64
A_C               object
ADDRESS           object
BOOK              object
BG                 int64
WALK_SEQ          object
METER_NO          object
ME_PHASE         float64
S_LOAD           float64
C_LOAD           float64
METER_COND       float64
OMF               object
CATAGORY          object
BILL_MONTH         int64
C_SR_RED          object
C_OFPK_RED        object
C_PK_RED          object
TO_UNIT           object
CUR_M_PRIN        object
CUR_LPS           object
CUR_VAT           object
CUR_M_BILL        object
PAY_DATE          object
PRIN_PAID         object
VAT_PAID          object
TOTAL_ADJ         object
PRN_BALANCE       object
VAT_BALANCE       object
LPS_BALANCE       object
TOTAL_BALANCE     object
TOTAL_BILL        object
dtype: object

In [6]:
df.describe()

CUST_NUM            BG      ME_PHASE        S_LOAD        C_LOAD  \
count  1.671400e+04  16714.000000  14726.000000  16714.000000  16714.000000   
mean   8.076452e+07      7.820330      1.028657      2.436341      2.436101   
std    6.550674e+06      3.483698      0.237689      9.294850      9.294854   
min    7.590013e+07      2.000000      1.000000      1.000000      1.000000   
25%    7.595106e+07      4.000000      1.000000      2.000000      2.000000   
50%    7.675105e+07      8.000000      1.000000      2.000000      2.000000   
75%    9.036187e+07     10.000000      1.000000      2.000000      2.000000   
max    9.126218e+07     22.000000      3.000000    800.000000    800.000000   

         METER_COND  BILL_MONTH  
count  14726.000000     16714.0  
mean       0.363439    202505.0  
std        2.142415         0.0  
min        0.000000    202505.0  
25%        0.000000    202505.0  
50%        0.000000    202505.0  
75%        0.000000    202505.0  
max       13.000000    202505.0

In [7]:
# Check for missing values
df.isnull().sum()

LOC                  0
CUST_NAM             0
F_H_NAME         16380
CUST_NUM             0
A_C                  1
ADDRESS              0
BOOK                 0
BG                   0
WALK_SEQ             0
METER_NO             0
ME_PHASE          1988
S_LOAD               0
C_LOAD               0
METER_COND        1988
OMF                  0
CATAGORY             0
BILL_MONTH           0
C_SR_RED          3024
C_OFPK_RED       16691
C_PK_RED         16649
TO_UNIT              0
CUR_M_PRIN           0
CUR_LPS              0
CUR_VAT              0
CUR_M_BILL           0
PAY_DATE          8485
PRIN_PAID            0
VAT_PAID             0
TOTAL_ADJ            0
PRN_BALANCE          0
VAT_BALANCE          0
LPS_BALANCE          0
TOTAL_BALANCE        0
TOTAL_BILL           0
dtype: int64

Missing Value Summary:

| Column       | Missing Values | Comment                                 |
| ------------ | -------------- | --------------------------------------- |
| `C_SR_RED`   | 3,024          | Moderate missing (usable)               |
| `C_OFPK_RED` | 16,691         | High missing (\~majority)               |
| `C_PK_RED`   | 16,649         | High missing (\~majority)               |
| `METER_COND` | 1,988          | Moderate, might be categorical          |
| `ME_PHASE`   | 1,988          | Moderate, likely numerical |
    F_H_NAME         16380          not much important

In [8]:
df.columns

Index(['LOC', 'CUST_NAM', 'F_H_NAME', 'CUST_NUM', 'A_C', 'ADDRESS', 'BOOK',
       'BG', 'WALK_SEQ', 'METER_NO', 'ME_PHASE', 'S_LOAD', 'C_LOAD',
       'METER_COND', 'OMF', 'CATAGORY', 'BILL_MONTH', 'C_SR_RED', 'C_OFPK_RED',
       'C_PK_RED', 'TO_UNIT', 'CUR_M_PRIN', 'CUR_LPS', 'CUR_VAT', 'CUR_M_BILL',
       'PAY_DATE', 'PRIN_PAID', 'VAT_PAID', 'TOTAL_ADJ', 'PRN_BALANCE',
       'VAT_BALANCE', 'LPS_BALANCE', 'TOTAL_BALANCE', 'TOTAL_BILL'],
      dtype='object')

In [ ]:
for col in ['C_SR_RED', 'C_OFPK_RED', 'C_PK_RED']:
    missing_count = df[col].isna().sum()
    total = len(df)
    print(f"{col}: {missing_count} missing ({(missing_count/total)*100:.2f}%)")
    


C_SR_RED: 3024 missing (18.09%)
C_OFPK_RED: 16691 missing (99.86%)
C_PK_RED: 16649 missing (99.61%)
